flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1115,6 +1117,16 @@ mlir::Value IntrinsicLibrary::genDim(mlir::Type resultType,
   return builder.create<mlir::SelectOp>(loc, cmp, diff, zero);
 }
 
+// DPROD
+mlir::Value IntrinsicLibrary::genDprod(mlir::Type resultType,
+				       llvm::ArrayRef<mlir::Value> args) {
+  assert(args.size() == 2);
+  assert(fir::isa_real(resultType) && (resultType.getKind() == mlir::StandardTypes::F64) && "Result must be double precision in DPROD");
```

Comment:
Looks like this may need to go through clang-format.

Reply:
Done.  Thanks!